doc/vlan/SONiC_vlan_translation_qnq.md

```diff
@@ -0,0 +1,1025 @@
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| C-VLAN                   | Customer VLAN               |
+| S-VLAN                   | Service Provider  VLAN               |
+| C-TAG                    | Customer VLAN Tag               |
+| S-TAG                    | Service Provider VLAN Tag.              |
+| CE                       | Customer Edge          |
+| PE                       | (Service) Provider Edge|
+| VXLAN                    | Virtual extensible Local Area Network  |
+| VTEP                     | VXLAN Tunnel End Point|
+| VNI                      | Virtual Network Identifier|
+
+# 1 Feature Overview
+
+This feature supports carrying customer traffic over Provider networks using QinQ Tunneling or VLAN translation mechanisms. 
+This feature also supports carrying customer traffic over a Provider Network using VXLAN. 
+This implementation will only support carrying Layer2 traffic across provider network.  
+
+A Q-in-Q VLAN tunnel will enable a Service Providers to seperate VLAN traffic from different customers in their network and also group multiple customer VLANs(C-VLAN) and tunnel them with a single service provider VLAN(S-VLAN). In Q-in-Q tunneling, customer traffic arriving at a Service Provider Edge (PE) interface will be added with an S-TAG, that identifies a single C-VLAN or a group of C-VLANs. These packets travel in the provider network based on the S-TAG while preserving the original customer VLAN (C-TAG) in the packet. Though the packet is double tagged, the forwarding decisions are made based on the outer S-TAG. At the other end of the provider network, as the traffic leaves PE, the S-VLAN in the packet is removed before handing off the traffic to the customer network.
+
+VLAN-translation will allow swapping single or double tagged traffic (C-VLANS) with an S-VLAN at the provider edge device and traffic will be forwarded based on S-VLAN in the provider network. The C-VLANs will be lost in the translation.
+
+If the Provider network enabled with VXLAN overlay, the customer traffic identified by the S-VLANs need to be mapped to VXLAN VNIs and is forwarded based on VNI. The S-VLAN header will be replaced with VNI. At the egress of the provider network, the VNI will be mapped to an S-VLAN and further forwarding decisions are made based on S-VLAN.
+
+This feature will be supported on Physical Interfaces (Ethernet) and PortChannels.
+
+## 1.1 Target Deployment Use Cases
+### 1.1.1 VLAN Translation usecases
+#### Usecase - 1  Native VLAN Translation
+![VLAN translate use case](./images/map_cvlan_native.png)
+
+#### Usecase - 2  VLAN Translation with VxLAN
+   * Ingress - Map single/double C-TAGs to VNI
+   * Egress  - Map VNI to single/double C-TAGs
+
+CVLANs on both the ends are same
+![VLAN translate use case with VxLAN](./images/Map_CVLAN_VNI.png)
+
+#### Usecase - 3 VLAN Translation with VxLAN
```

Comment:
could you please add step by step explanation for better understanding

Reply:
Section 1.3.1.1 captures these details